.idea/misc.xml

```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
Unfortunately this couldn't be renamed, there must be only one misc.xml in IntelliJ project, as far as I know. This is silly and cause many complains (you can google "misc xml git"), but there is no other way. What plugins do you use and what kind of info they store in this file?

Reply:
The conflict occurs because this file is added to git in this patch, but it already exists in my machine. 

> Unfortunately this couldn't be renamed, there must be only one misc.xml in IntelliJ project, as far as I know.

Then, I would like to be conservative about this issue. People may want to add their own options to this file which might be difficult to be shared and thus maintained by git.

Is it possible to move EntryPointsManager configuration to another file? I think it's the best option. If not possible, I suggest to not add it.

Reply:
> The conflict occurs because this file is added to git in this patch, but it already exists in my machine.

Yes, but once this file is checked in the repo, there shouldn't be more conflicts.

> People may want to add their own options to this file which might be difficult to be shared and thus maintained by git.

Note that this is per-project settings. The trade-off is to ask people not to use other per-project settings, when working specifically with the druid project, or else "share" them in PRs.

> Is it possible to move EntryPointsManager configuration to another file? I think it's the best option. If not possible, I suggest to not add it.

It's not possible. Not including it will not allow to track unused declarations using IntelliJ inspections CI.

Reply:
> The trade-off is to ask people not to use other per-project settings, when working specifically with the druid project, or else "share" them in PRs.

Hmm, I basically agree on that this default configuration can be useful for developers. However, I think it shouldn't be mandatory. This is because 1) this configuration is specific to only Intellij, so developers using other tools cannot benefit from this and 2) IDE-specific configurations should be able to be changed easily without worrying about PR review or code maintenance (like the SDK name issue below). 

It may be fine if this file is rarely changed, but I'm not sure which kind of configurations it can contain. Do you have any links to any document about this file? I couldn't find anything.

Maybe it's better to maintain IDE-specific configurations in some other places and let developers use or copy them. This can be done by documentation.

Reply:
This official answer suggests to check into VCS all files under `.idea` dir, with few exceptions, but `misc.xml` is not one of those, so it should be included: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems

Regarding other IDEs: I don't think we used to try to keep Eclipse and Idea settings/configs in sync, and Eclipse is lagging now, because less developers use it. Partially to mitigate this, IntelliJ inspections CI was introduced, and _this_ is why I'm checking misc.xml now.

Reply:
I see. The only thing I'm concerned with is frequent conflicts due to different per-developer settings, but it looks that per-developer settings are not stored in this file except the project SDK name. It sounds we can compromise for more efficient development and code review.